platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/BuildInvocationsBuilder.java

```diff
@@ -94,11 +95,9 @@ private DefaultProjectIdentifier getProjectIdentifier(Project project) {
 
     // build tasks without project reference
     private List<LaunchableGradleTask> tasks(Project project, DefaultProjectIdentifier projectIdentifier) {
-        List<LaunchableGradleTask> tasks = Lists.newArrayList();
-        for (Task task : taskLister.listProjectTasks(project)) {
-            tasks.add(buildFromTask(new LaunchableGradleTask(), projectIdentifier, task));
-        }
-        return tasks;
+        return taskLister.listProjectTasks(project).stream()
+            .map(task -> buildFromTask(new LaunchableGradleTask(), projectIdentifier, task))
+            .collect(toList());
```

Comment:
❌ Collecting a stream into a list with a `Collectors.toList` does not guarantee that the returned list is serializable (see javadoc of the method).
Since we are going to serialize the list, we need to make sure this won't break when running in future versions of Java. I presume it currently works, because current JDKs do return a serializable list.

I suggest to not fix it here, but instead make an immutable copy of the task list when we assign it into the field here:
https://github.com/gradle/gradle/blob/1295becbd5d1c33eb88d9baf8153a8073a7c4ee0/platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/model/DefaultBuildInvocations.java#L43
->
https://github.com/gradle/gradle/blob/35a9b833a1185c500bb9b90379aebe3a50a63875/platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/model/DefaultGradleProject.java#L128